core/modules/comment/comment.module

```diff
@@ -135,7 +151,7 @@ function comment_field_extra_fields() {
   $return = array();
 
   foreach (node_type_get_types() as $type) {
-    if ($type->settings['comment_subject_field']) {
+    if ($type->settings['comment_subject_options'] == COMMENT_TITLE_AUTO) {
```

Comment:
Hm. To my understanding comment_subject_field would be same as COMMENT_TITLE_CUSTOM...

Reply:
@indigoxela I'm not sure if I understand your point, but maybe this will help explain what I did. That doesn't mean that what I did is correct or optimal.

comment_subject_field was a boolean setting with titles either 'auto generated' = false or 'custom" = true.

I replaced comment_subject_field with a config setting comment_subject_options that has three options.

1) COMMENT_TITLE_CUSTOM
2) COMMENT_TITLE_AUTO
3) COMMENT_TITLE_HIDDEN

Does this help? Am I missing something?